eclipse-sirius/sirius-web
# [2942] Add the Card DnD capability on Deck

- Make it possible to specify the card drop behavior on a Lane in the Deck view model.
- Add the DropCard mutation to handle the front-end drag and drop and execute the tool.

Bug: https://github.com/eclipse-sirius/sirius-web/issues/2942

# Pull request template

## General purpose
What is the main goal of this pull request?

## Project management

## Architectural decision records (ADR)

## Dependencies

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- Hooks are properly typed:

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose

### Architecture

## Review

### How to test this PR?

- Create a new Blank Studio and create a Deck view by using the deck stereotype
- Create a new Task project from the template
- Create a new Deck Representation on the Daily Dev Project
- Try to DnD a card within a Lane or to an other lane: The ordering should be persisted and reflected in the model
